src/plugins/plugin.legend.js

```diff
@@ -237,26 +237,54 @@ var Legend = Element.extend({
 			minSize.height = me.maxHeight; // fill all the height
 		}
 
+		var getLineWidths = function(boxWidth, textLines) {
+			return textLines.map(function(textLine) {
+				return boxWidth + (fontSize / 2) + ctx.measureText(textLine).width;
+			});
+		};
+
 		// Increase sizes here
 		if (display) {
 			ctx.font = labelFont.string;
 
 			if (isHorizontal) {
+
 				// Labels
 
 				// Width of each line of legend boxes. Labels wrap onto multiple lines when there are too many to fit on one
 				var lineWidths = me.lineWidths = [0];
+				var maxHeight = 0;
 				var totalHeight = 0;
 
 				ctx.textAlign = 'left';
 				ctx.textBaseline = 'top';
 
 				helpers.each(me.legendItems, function(legendItem, i) {
 					var boxWidth = getBoxWidth(labelOpts, fontSize);
-					var width = boxWidth + (fontSize / 2) + ctx.measureText(legendItem.text).width;
+					var width;
+					var height;
+					var grossHeight;
+
+					if (helpers.isArray(legendItem.text)) {
```

Comment:
Could you do something like:

    var legendText = helpers.isArray(legendItem.text) ? legendItem.text : (helpers.isNullOrUndef(legendItem.text) ? [] : [legendItem.text]);

And then just use that variable to remove this if/else check and only handle the array case?

Reply:
That is ugly code imo, I'd do that but for the record, I'm against that.

Reply:
Would this be any better?
```
var legendText = legendItem.text;
if (!helpers.isArray(legendText)) {
    legendText = helpers.isNullOrUndef(legendText) ? [] : [legendText];
}
```

I wasn't sure if you were against the multiple ternary operators or handling everything as an array

Reply:
Well, not sure this twisting of the code is necessary, it doesn't add anything, just reduces clarity...

Reply:
There are four places where a check `if (helpers.isArray(legendItem.text))` is done. I think you could remove three of those checks (i.e. all except the strikethrough code) if you handle everything as an array instead of needing separate cases for whether there's a single item or multiple items

Reply:
I don't think such an endeavor would provide any benefit whatsoever, it might shorten code, but it won't increase clarity or maintainability.

Reply:
Feel free to refactor the code after accepting the PR. The essence of the PR is to render multiline labels properly, and it does with code that is good enough (after many iterations).